packages/flutter/lib/src/rendering/animated_size.dart

```diff
@@ -0,0 +1,132 @@
+    Curve curve,
+    RenderBox child,
+    FractionalOffset alignment: FractionalOffset.center,
+    VoidCallback callback
+  }) : super(child: child, alignment: alignment) {
+    _controller = new AnimationController(
+      duration: duration
+    );
+    _animation = new CurvedAnimation(
+      parent: _controller,
+      curve: curve
+    );
+    _callback = callback;
+  }
+
+  AnimationController _controller;
+  CurvedAnimation _animation;
+  Size _sourceSize;
+  Size _targetSize;
+  bool _lastFrameChanged = false;
+  bool _hasVisualOverflow;
+
+  VoidCallback get callback => _callback;
+  VoidCallback _callback;
+  set callback(VoidCallback value) {
+    if (value == _callback)
+      return;
+    _callback = value;
+  }
+
+  /// The duration of the animation.
+  Duration get duration => _controller.duration;
+  set duration(Duration value) {
+    assert(value != null);
+    if (value == _controller.duration)
+      return;
+    _controller.duration = value;
+  }
+
+  /// The curve of the animation.
+  Curve get curve => _animation.curve;
+  set curve(Curve value) {
+    assert(value != null);
+    if (value == _animation.curve)
+      return;
+    _animation.curve = value;
+  }
+
+  @override
+  void detach() {
+    _controller?.stop();
+    super.detach();
+  }
+
+  void _trigger() {
+    _controller
+      ..removeListener(markNeedsLayout)
+      ..value = 0.0
+      ..forward()
+      ..addListener(markNeedsLayout);
+  }
+
+  Size get _animatedSize {
+    return Size.lerp(_sourceSize, _targetSize, _animation.value);
+  }
+
+  @override
+  void performLayout() {
+    if (_callback != null)
+      _callback();
+
+    _hasVisualOverflow = false;
+
+    if (child != null) {
+      child.layout(constraints, parentUsesSize: true);
+      final Size childSize = constraints.constrain(child.size);
+
+      if (_targetSize != childSize) {
+        _sourceSize = _animatedSize ?? childSize;
+        _targetSize = childSize;
+
+        if (_lastFrameChanged) {
+          size = constraints.constrain(_targetSize);
+          _controller.stop();
+        } else {
+          // Don't register first change (i.e. when _targetSize == _sourceSize)
+          // as a last-frame change.
+          if (_targetSize != _sourceSize)
+            _lastFrameChanged = true;
+
+          _trigger();
+
+          size = constraints.constrain(_animatedSize);
+        }
+      } else {
+        _lastFrameChanged = false;
+
+        size = constraints.constrain(_animatedSize);
+      }
+
+      alignChild();
+
+      if (size.width < _targetSize.width || size.height < _targetSize.height)
+        _hasVisualOverflow = true;
+    } else {
+      size = _sourceSize = _targetSize = constraints.constrain(Size.zero);
+    }
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    if (child != null && _hasVisualOverflow) {
+      final Rect rect = Point.origin & size;
+      context.pushClipRect(needsCompositing, offset, rect, super.paint);
+    }
```

Comment:
If there's no visual overflow, you still want to call super.paint, just without the clip.  What you have here won't paint the child.